### PR TITLE
Improve naming and descriptions

### PR DESCRIPTION
**Pull request by olliahonen**
* Server or quantum computer instead of backend
* Other improvements to grammar and descriptions
* No changes to functionality

Closes COMP-124

### CODE REVIEW

File: src/cirq_iqm/iqm_remote.py [hunk 1]
```diff
@@ -66,12 +66,12 @@ def run_sweep(
             params: cirq.Sweepable,
             repetitions: int = 1,
     ) -> list[cirq.Result]:
-        """Samples from the given quantum circuit.
+        """Samples from, i.e. executes, the given quantum circuit.
```
Comment (manevala) [inline src/cirq_iqm/iqm_remote.py:69]: "Samples from the given quantum circuit." comes directly from the the parent class, not sure if changing it brings more clarity.

File: src/cirq_iqm/iqm_remote.py [hunk 2]
```diff
@@ -66,12 +66,12 @@ def run_sweep(
...
 
         Nontrivial sweeping is not yet supported, ``params`` should be ``None``.
 
         Args:
-            program: The circuit to sample from.
+            program: The circuit to execute.
```
Comment (manevala) [inline src/cirq_iqm/iqm_remote.py:74]: Same as above. "The circuit to sample from" comes from the parent class.

**Review by manevala — APPROVED** [review]

File: examples/demo_iqm_engine.py [hunk 3]
```diff
@@ -24,7 +24,7 @@
 from cirq_iqm.iqm_remote import IQMSampler
 
 
-def demo_adonis_backend():
+def demo_adonis_run():
     """
     Run a circuit on the Adonis quantum computer.
```
Comment (smite) [inline examples/demo_iqm_engine.py:29]:
```suggestion
    Run a circuit on an Adonis quantum computer.
```

File: src/cirq_iqm/iqm_client.py [hunk 4]
```diff
@@ -74,7 +74,7 @@ class SingleQubitMappingDTO(BaseModel):
 
 
 class RunRequestDTO(BaseModel):
-    """Request for IQM backend to execute a circuit.
+    """Request for IQM quantum computer to execute a circuit.
```
Comment (smite) [inline src/cirq_iqm/iqm_client.py:77]:
```suggestion
    """Request for IQM quantum computing service to execute a circuit.
```

File: src/cirq_iqm/iqm_client.py [hunk 5]
```diff
@@ -108,11 +108,11 @@ def from_dict(inp: dict[str, Union[str, dict]]) -> RunResult:
         return RunResult(status=RunStatus(input_copy.pop("status")), **input_copy)
 
 
-class IQMBackendClient:
-    """Provides access to a remote IQM quantum device.
+class IQMClient:
+    """Provides access to an IQM quantum computer.
```
Comment (smite) [inline src/cirq_iqm/iqm_client.py:112]:
```suggestion
    """Provides access to quantum computers through the IQM computation control service.
```

File: src/cirq_iqm/iqm_client.py [hunk 6]
```diff
@@ -108,11 +108,11 @@ def from_dict(inp: dict[str, Union[str, dict]]) -> RunResult:
...
 
     Args:
-        url: Endpoint for accessing the device. Has to start with http or https.
+        url: Endpoint for accessing the server. Has to start with http or https.
```
Comment (smite) [inline src/cirq_iqm/iqm_client.py:115]:
```suggestion
        url: Endpoint for accessing the service. Has to start with http or https.
```

File: src/cirq_iqm/iqm_client.py [hunk 7]
```diff
@@ -125,7 +125,7 @@ def submit_circuit(
             qubit_mapping: list[SingleQubitMappingDTO],
             shots: int = 1
     ) -> UUID:
-        """Submits a quantum circuit to be executed on the backend.
+        """Submits a quantum circuit to be executed on the quantum computer.
```
Comment (smite) [inline src/cirq_iqm/iqm_client.py:128]:
```suggestion
        """Submits a quantum circuit to be executed on a quantum computer.
```

File: src/cirq_iqm/iqm_remote.py [hunk 8]
```diff
@@ -49,15 +49,15 @@ def _serialize_iqm(circuit: cirq.Circuit) -> CircuitDTO:
...
 
 class IQMSampler(cirq.work.Sampler):
-    """Circuit sampler for evaluating quantum circuits on IQM quantum devices.
+    """Circuit sampler for executing quantum circuits on IQM quantum computers.
 
     Args:
-        url: Endpoint for accessing the device. Has to start with http or https.
-        settings: Settings for the quantum hardware.
-        qubit_mapping: A dict that maps qubit logical names to physical qubit names
+        url: Endpoint for accessing the server interface. Has to start with http or https.
```
Comment (smite) [inline src/cirq_iqm/iqm_remote.py:55]:
```suggestion
        url: Endpoint for accessing the computation service. Has to start with http or https.
```

File: src/cirq_iqm/iqm_remote.py (hunk 1 above)
Comment (smite) [reply src/cirq_iqm/iqm_remote.py:69]: Let's keep it as it was. The user won't call this method directly anyway.

File: src/cirq_iqm/iqm_remote.py [hunk 9]
```diff
@@ -93,17 +93,17 @@ def _send_circuit(
             circuit: cirq.Circuit,
             repetitions: int = 1,
     ) -> cirq.study.Result:
-        """Sends the circuit to be executed by the remote IQM device.
+        """Sends the circuit to be executed on the IQM quantum computer.
```
Comment (smite) [inline src/cirq_iqm/iqm_remote.py:96]:
```suggestion
        """Sends the circuit to be executed by the IQM computation service.
```

File: src/cirq_iqm/iqm_remote.py (hunk 1 above)
Comment (olliahonen) [reply src/cirq_iqm/iqm_remote.py:69]: AFAIK this is supposed to be one of the functions the user may call directly. But anyway, no point in copying the superclass docstring verbatim, I'll just delete it here. IDEs and our docs generation are able to pick up the docstring from the superclass.

Comment (olliahonen) [reply src/cirq_iqm/iqm_remote.py]: Docstring removed here.

File: src/cirq_iqm/iqm_remote.py (hunk 2 above)
Comment (olliahonen) [reply src/cirq_iqm/iqm_remote.py]: Docstring removed here.

File: src/cirq_iqm/iqm_remote.py (hunk 1 above)
Comment (olliahonen) [reply src/cirq_iqm/iqm_remote.py]: Turns out the docstring was not verbatim copy of the superclass one.

Comment (olliahonen) [reply src/cirq_iqm/iqm_remote.py]: Reworded.

File: src/cirq_iqm/iqm_remote.py (hunk 9 above)
Comment (olliahonen) [reply src/cirq_iqm/iqm_remote.py:94]: I'll remove the controversial part. It reads just as well like that :) Underscore method, after all.

File: src/cirq_iqm/iqm_client.py (hunk 5 above)
Comment (olliahonen) [reply src/cirq_iqm/iqm_client.py:112]: I see your point here but it's not so easy to keep it consistent across all the modules... It seems that "server/server interface" and "quantum computer" is the least confusing way to explain what we have here. Additional concepts such as a service, however real, only seem to complicate the documentation.

Also, we should be proud of our main product at IQM, the quantum computers :)

**Pull request merged**